communitiesuk/funding-service-design-frontend
# Fs 2877 cached fund round

https://dluhcdigital.atlassian.net/browse/FS-2877

### Change description
- Implemented LRU caching for `get_fund_data` & `get_round_data` functions. This reduces the API calls to fund store & improves the performance
- Caching timeout is set to 300 secs (=5 mins)
- Unit tests added for the LRU caching feature.
- Reduce duplicate and redundant code that does the same thing with the same error handling/fallbacks

### How to test
- Put breakpoints in the get_fund and get_round methods of fund_store
- Load the frontend in a browser and check that the above breakpoints don't get hit except on the first retrieval of a session (ie the frontend has cached the data)


### Screenshots of UI changes (if applicable)
n/a
